File: spring_server/src/main/java/com/semothon/spring_server/room/entity/RoomInterest.java
package com.semothon.spring_server.room.entity;

import com.semothon.spring_server.interest.entity.Interest;
import com.semothon.spring_server.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
@ToString(of = {"roomInterestId"})
@Table(name = "room_interests",
        indexes = {
        },
        uniqueConstraints = {
                @UniqueConstraint(columnNames = {"room_id", "interest_id"})
        }
)
public class RoomInterest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long roomInterestId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id", nullable = false)
    private Room room;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "interest_id", nullable = false)
    private Interest interest;

    public void updateRoom(Room room){
        this.room = room;
        room.addRoomInterest(this);
    }

    public void updateInterest(Interest interest){
        this.interest = interest;
        interest.addRoomInterest(this);
    }
}

File: fastapi_server/app/models/room_interest.py
from sqlalchemy import Column, ForeignKey, Integer, UniqueConstraint
from sqlalchemy.orm import relationship
from database import Base

class RoomInterest(Base):
    __tablename__ = "room_interests"
    __table_args__ = (
        UniqueConstraint("room_id", "interest_id", name="uq_room_interest"),
    )

    room_interest_id = Column(Integer, primary_key=True, autoincrement=True)
    room_id = Column(Integer, ForeignKey("rooms.room_id"), nullable=False)
    interest_id = Column(Integer, ForeignKey("interests.interest_id"), nullable=False)

    room = relationship("Room")
    interest = relationship("Interest")